lib/http.ex

```diff
@@ -166,4 +163,9 @@ defmodule Braintree.HTTP do
   for {code, status} <- @statuses do
     def code_to_reason(unquote(code)), do: unquote(status)
   end
+
+  defp resolve_unprocessable_entity_error(%{"api_error_response" => api_error_response}),
+    do: Error.new(api_error_response)
+  defp resolve_unprocessable_entity_error(%{"unprocessable_entity" => _}),
+    do: :unprocessable_entity
```

Comment:
Really this is multiple error types, it could simply be `resolve_error_response`.

Let's go with `Error.new(%{message: "Unprocessable Entity"})` to return something consistent. Also, please use `do/end` here.